java/src/main/java/com/logicalclocks/hsfs/engine/SparkEngine.java

```diff
@@ -93,6 +95,53 @@ private SparkEngine() {
     sparkSession.conf().set("hive.exec.dynamic.partition.mode", "nonstrict");
     // force Spark to fallback to using the Hive Serde to read Hudi COPY_ON_WRITE tables
     sparkSession.conf().set("spark.sql.hive.convertMetastoreParquet", "false");
+
+    validateSparkConfiguration();
+  }
+
+  @SneakyThrows
+  private void validateSparkConfiguration() {
+    String key = "spark.hadoop.hops.ssl.trustore.name";
+    if (sparkSession.conf().get(key).isEmpty())
+      throw new FeatureStoreException("Invalid: " + key);
+    key = "spark.hadoop.hops.rpc.socket.factory.class.default";
+    if (!sparkSession.conf().get(key)
+            .equals("io.hops.hadoop.shaded.org.apache.hadoop.net.HopsSSLSocketFactory"))
+      throw new FeatureStoreException("Invalid: " + key);
+    key = "spark.serializer";
+    if (!sparkSession.conf().get(key)
+            .equals("org.apache.spark.serializer.KryoSerializer"))
+      throw new FeatureStoreException("Invalid: " + key);
+    key = "spark.hadoop.hops.ssl.hostname.verifier";
+    if (!sparkSession.conf().get(key)
+            .equals("ALLOW_ALL"))
+      throw new FeatureStoreException("Invalid: " + key);
+    key = "spark.hadoop.hops.ssl.keystore.name";
+    if (sparkSession.conf().get(key).isEmpty())
+      throw new FeatureStoreException("Invalid: " + key);
+    key = "spark.hadoop.fs.hopsfs.impl";
+    if (!sparkSession.conf().get(key)
+            .equals("io.hops.hopsfs.client.HopsFileSystem"))
+      throw new FeatureStoreException("Invalid: " + key);
+    key = "spark.hadoop.hops.ssl.keystores.passwd.name";
+    if (sparkSession.conf().get(key).isEmpty())
+      throw new FeatureStoreException("Invalid: " + key);
+    key = "spark.hadoop.hops.ipc.server.ssl.enabled";
+    if (!sparkSession.conf().get(key)
+            .equals("true"))
+      throw new FeatureStoreException("Invalid: " + key);
+    //todo should jars path be compared?
```

Comment:
No I think it shouldn't, on databricks we set it, but users might use their own spark setup and have it in some other location.